### PR TITLE
Fix bug on wait_for so that timeouts remove it from the waiters_set

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/impl/event_manager_base.py [hunk 1]
```diff
@@ -272,7 +272,11 @@ async def wait_for(
         pair = (predicate, future)
 
         waiter_set.add(pair)  # type: ignore[arg-type]
-        return await asyncio.wait_for(future, timeout=timeout)
+        try:
+            return await asyncio.wait_for(future, timeout=timeout)
+        except Exception:
+            waiter_set.remove(pair)  # type: ignore[arg-type]
```
Comment (FasterSpeeding) [inline hikari/impl/event_manager_base.py:278]: i assume this won't be removed anywhere else right?

Comment (davfsa) [reply hikari/impl/event_manager_base.py:278]: It does, so changed it so that its only removed on Timeout instead of all exceptions.

**Review by FasterSpeeding — APPROVED** [review]
I assume that on a timeout error specifically it won't have been removed anywhere else

**Comment by davfsa** [comment]
> 
> 
> I assume that on a timeout error specifically it won't have been removed anywhere else

Yep, that is the issue this pr fixes 😄

**Pull request merged**